### ISSUE TITLE
8.2.1 模型装配  依照您的代码compile    训练出来的准确率几乎为0

### ISSUE THREAD

**Issue by zhaowei0315** [dragen1860/Deep-Learning-with-TensorFlow-book#140]
依照您的代码    训练出来的准确率几乎为0

network.compile(
                optimizer=optimizers.Adam(learning_rate=0.01),# 采用Adam 优化器，学习率为0.01  
                loss=losses.CategoricalCrossentropy(from_logits=True),# 采用交叉熵损失函数，包含Softmax  
                metrics=['accuracy'] # 设置测量指标为准确率
                )

使用如下设置，准确率就非常不错

network.compile(
                optimizer='adam',
                loss='sparse_categorical_crossentropy',
                metrics=['accuracy']
                )

请问这两种方式有什么不一样吗？
我将第一种方式中的参数删掉，效果还是一样的不理想。

**Comment by dragen1860** [comment]
你好，这个错误在最新版已经修正了，Sparse的意思就是内部实现了one hot编码，如果使用第一种方式，则需要在预处理函数中one hot变换。
